aptos-move/framework/aptos-framework/sources/vesting_without_staking.move

```diff
@@ -0,0 +1,1132 @@
+        assert!(period_duration > 0, error::invalid_argument(EZERO_VESTING_SCHEDULE_PERIOD));
+        assert!(
+            start_timestamp_secs >= timestamp::now_seconds(),
+            error::invalid_argument(EVESTING_START_TOO_SOON),
+        );
+
+        VestingSchedule {
+            schedule,
+            start_timestamp_secs,
+            period_duration,
+            last_vested_period: 0,
+        }
+    }
+
+    /// Create a vesting contract with a given configurations.
+    public fun create_vesting_contract(
+        admin: &signer,
+        buy_ins: SimpleMap<address, Coin<AptosCoin>>,
+        vesting_schedule: VestingSchedule,
+        withdrawal_address: address,
+        contract_creation_seed: vector<u8>,
+    ): address acquires AdminStore {
+        assert!(
+            !system_addresses::is_reserved_address(withdrawal_address),
+            error::invalid_argument(EINVALID_WITHDRAWAL_ADDRESS),
+        );
+        assert_account_is_registered_for_apt(withdrawal_address);
+        let shareholders_address = &simple_map::keys(&buy_ins);
+        assert!(vector::length(shareholders_address) > 0, error::invalid_argument(ENO_SHAREHOLDERS));
+
+        let shareholders = simple_map::create<address, Amount>();
+        let grant = coin::zero<AptosCoin>();
+        let grant_amount = 0;
+        vector::for_each_ref(shareholders_address, |shareholder| {
+            let shareholder: address = *shareholder;
+            let (_, buy_in) = simple_map::remove(&mut buy_ins, &shareholder);
+            let init = coin::value(&buy_in);
+            coin::merge(&mut grant, buy_in);
+            simple_map::add(&mut shareholders, shareholder, Amount {
+                init_amount: init,
+                left_amount: init,
+            });
+            grant_amount = grant_amount + init;
+        });
+        assert!(grant_amount > 0, error::invalid_argument(EZERO_GRANT));
+
+        // If this is the first time this admin account has created a vesting contract, initialize the admin store.
+        let admin_address = signer::address_of(admin);
+        if (!exists<AdminStore>(admin_address)) {
+            move_to(admin, AdminStore {
+                vesting_contracts: vector::empty<address>(),
+                nonce: 0,
+                create_events: new_event_handle<CreateVestingContractEvent>(admin),
+            });
+        };
+
+        // Initialize the vesting contract in a new resource account. This allows the same admin to create multiple
+        // pools.
+        let (contract_signer, contract_signer_cap) = create_vesting_contract_account(admin, contract_creation_seed);
+        let contract_signer_address = signer::address_of(&contract_signer);
+        coin::deposit(contract_signer_address, grant);
+
+        // Add the newly created vesting contract's address to the admin store.
+        let contract_address = signer::address_of(&contract_signer);
```

Comment:
isn't it possible to reuse `contract_signer_address` which already contains the address? Does coin::deposit create a `borrow` reference issue?

Reply:
Removed for the first part. Why did deposit create a borrow reference issue here? I did not get it.

Reply:
Thanks!